docs/neural_modeling/plot_05_place_cells.py

```diff
@@ -0,0 +1,410 @@
+
+plt.tight_layout()
+
+# %%
+# ## Model selection
+#
+# While this model captures nicely the features-rate relationship, it is not necessarily the simplest model. Let's construct several models and evaluate their score to determine the best model.
+#
+# !!! note "To shorten this notebook, only a few combination are tested. Feel free to expand this list."
+#
+
+models = {
+    "position": position_basis,
+    "position + speed": position_basis + speed_basis,
+    "position + phase": position_basis + phase_basis,
+    "position + phase + speed": position_basis + phase_basis + speed_basis,
+}
+features = {
+    "position": (position,),
+    "position + speed": (position, speed),
+    "position + phase": (position, theta),
+    "position + phase + speed": (position, theta, speed),
+}
+
+# %%
+# In a loop, we can (1) evaluate the basis, (2), fit the model, (3) compute the score and (4) predict the firing rate. For evaluating the score, we can define a train set of intervals and a test set of intervals.
+
+train_iset = position.time_support[
+    0 : len(position.time_support) // 2
+]  # Taking the first half of the recording as the train set
+test_iset = position.time_support[
+    len(position.time_support) // 2 :
+]  # Taking the second half of the recording as the test set
+
+# %%
+# Let's train all the models.
+scores = {}
+predicted_rates = {}
+
+for m in models:
+    print("1. Evaluating basis : ", m)
+    X = models[m].evaluate(*features[m])
+
+    print("2. Fitting model : ", m)
+    # glm = nmo.glm.GLM()
+    glm.fit(
+        X.restrict(train_iset)[:, np.newaxis, :],
+        count.restrict(train_iset)[:, np.newaxis],
+    )
+
+    print("3. Scoring model : ", m)
+    scores[m] = glm.score(
+        X.restrict(test_iset)[:, np.newaxis, :],
+        count.restrict(test_iset)[:, np.newaxis],
+        score_type="pseudo-r2-McFadden",
+    )
+
+    print("4. Predicting rate")
+    predicted_rates[m] = glm.predict(X.restrict(test_iset)[:, np.newaxis, :]) / bin_size
+
+
+scores = pd.Series(scores)
+scores = scores.sort_values()
+
+# %%
+# Let's compute scores for each models.
+
+plt.figure(figsize=(5, 3))
+plt.barh(np.arange(len(scores)), scores)
+plt.yticks(np.arange(len(scores)), scores.index)
+plt.ylabel("Pseudo r2")
+plt.tight_layout()
+
+
+# %%
+# Some models are clearly doing better than others.
+#
+# !!! warning " A proper model comparison should be done by scoring models repetitively on various train and test set. Here we are only doing partial models comparison for the sake of conciseness. "
+#
+# Alternatively, we can plot some tuning curves to compare each models visually.
+
+tuning_curves = {}
+
+for m in models:
+    tuning_curves[m] = {
+        "position": nap.compute_1d_tuning_curves_continuous(
+            predicted_rates[m], position, 50
+        ),
+        "speed": nap.compute_1d_tuning_curves_continuous(
+            predicted_rates[m], speed, 30, minmax=(0, 100)
+        ),
+    }
+
+
+def plot_tuning_curves(axis, tc, m):
```

Comment:
This probably goes into the utils too. you'll need to pass all the arguments, it is using some globals which would be  confusing if a person tries to use it outside the script.

Reply:
Done